src/media/inner.rs

```diff
@@ -543,6 +551,15 @@ impl MediaInner {
                 len
             }
         };
+        let pt = next.pt;
+        let seq_no = next.seq_no;
+        if matches!(next.body, NextPacketBody::Regular { .. }) {
+            if let Some(buffer_tx) = self.buffers_tx.get_mut(&pt) {
+                if let Some(pkt) = buffer_tx.take_next(now) {
```

Comment:
Changing `take_next` from `Packetized` to `Option<Packetized>` introduces the possibility of an incorrect state.

Since `maybe_packetized` returns `Option<&Packetized>`, we get a `Some(packet)` back when there is a packet to send. If `maybe_next` returns `Some` but the following `take_next` returns `None`, something is seriously wrong with the logic. This `if let Some(pkt)` masks that potential problem.

I think we should either continue with `take_next` returning a non-optional `Packetized`, or do an `.expect()` force unwrap here. Either way we want a panic for this inconsistency.

Another solution: Since the `rtx_cache` is an internal detail to `PacketizingBuffer`, we could also move the `packetized` from one buffer to the other inside `take_next`, avoiding this state inconsistency. We would then retain `&Packetized` signature of `take_next`.

Reply:
Yes, you have a good point.  I originally tried to do the "take_next moves between buffers" thing, but found it awkward because it provides a seq_no when taking.  But I guess if I called it "cache_next_as_sent" or something like that it could work.

I think a better approach would be to do what I did in #203: have maybe_next be take_next instead.  I was trying to minimize the diff here, but maybe I want too far.  Let me take a shot at going that direction and see how it goes.

Reply:
OK, I tried that route and it seems fine, except for a little duplicate code between RTX and non-RTX packets.  Let me know what you think.